tests/functional/metadata/mongodb/listObject.spec.js

```diff
@@ -0,0 +1,306 @@
+            maxKeys: 100,
+        };
+
+        return metadata.listObject(bucketName, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert(data);
+            assert.strictEqual(data.Contents.length, 0);
+            return done();
+        });
+    });
+
+    it('Should list all versions of objects', done => {
+        const bucketName = BUCKET_NAME;
+        const params = {
+            listingType: 'DelimiterVersions',
+            maxKeys: 1000,
+        };
+        return metadata.listObject(bucketName, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.Versions.length, 15);
+            return done();
+        });
+    });
+
+    it('Should truncate list of master versions of objects', done => {
+        const bucketName = BUCKET_NAME;
+        const params = {
+            listingType: 'DelimiterVersions',
+            maxKeys: 5,
+        };
+
+        return metadata.listObject(bucketName, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.Versions.length, 5);
+            return done();
+        });
+    });
+
+    it('Should list versions of objects that start with prefix', done => {
+        const bucketName = BUCKET_NAME;
+        const params = {
+            listingType: 'DelimiterVersions',
+            maxKeys: 100,
+            prefix: 'pfx2',
+        };
+
+        return metadata.listObject(bucketName, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.Versions.length, 5);
+            return done();
+        });
+    });
+
+    it('Should return empty results when bucket not existing (version)', done => {
+        const bucketName = 'non-existent-bucket';
+        const params = {
+            listingType: 'DelimiterVersions',
+            maxKeys: 100,
+        };
+
+        return metadata.listObject(bucketName, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert(data);
+            assert.strictEqual(data.Versions.length, 0);
+            return done();
+        });
+    });
+
+    it('should check entire list with pagination (version)', done => {
+        const versionsPerKey = {};
+        const bucketName = BUCKET_NAME;
+        const get = (maxKeys, keyMarker, versionIdMarker, cb) => metadata.listObject(bucketName, {
+            listingType: 'DelimiterVersions',
+            maxKeys,
+            keyMarker,
+            versionIdMarker,
+        }, logger, (err, res) => {
+            if (err) {
+                return cb(err);
+            }
+            res.Versions.forEach(version => {
+                if (versionsPerKey[version.key]) {
+                    versionsPerKey[version.key] += 1;
+                } else {
+                    versionsPerKey[version.key] = 1;
+                }
```

Comment:
Could be simplified:

```suggestion
                versionsPerKey[version.key] = (versionsPerKey[version.key] || 0) + 1
```

Reply:
fixed in https://github.com/scality/Arsenal/pull/1729/commits/a6f48b9f342344e6df5ca2e3b7a2d03d8737fbba